src/rpp/rpp/operators/element_at.hpp

```diff
@@ -0,0 +1,99 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2023 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/operators/fwd.hpp>
+
+#include <rpp/defs.hpp>
+#include <rpp/operators/details/strategy.hpp>
+
+#include <cstddef>
+
+namespace rpp::operators::details
+{
+    template<rpp::constraint::observer TObserver>
+    struct element_at_observer_strategy
+    {
+        using preferred_disposable_strategy = rpp::details::observers::none_disposable_strategy;
+
+        RPP_NO_UNIQUE_ADDRESS TObserver observer;
+        size_t                          index{};
+        mutable size_t                  current{};
```

Comment:
i think, it is possible to use 1 index like in take instead
 https://github.com/victimsnino/ReactivePlusPlus/blob/v2/src/rpp/rpp/operators/take.hpp#L28-L41

Reply:
I am not too sure for the on_error condition, just checking if `index == 0` doesn't work as it is decremented on each on_next. Like a sequence of n values with `element_at(n)`  will never error

Reply:
you can use index +1 as initial point, make -= 1 during on_next and emit value when zero was reached. In this case zero means "satisfied", but nvm, i can change it later

Reply:
Yeah though about this but makes logic a bit more complicated no?

Reply:
yeah, it is why you can skip it